### PR TITLE
CLN: de-duplicate reindex/align code

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
> FAILED pandas/tests/reductions/test_reductions.py::TestSeriesReductions::test_empty_multi[sum-0.0]

on AM

**Comment by jbrockmendel** [comment]
unrelated ResourceWarning

**Comment by jreback** [comment]
> unrelated ResourceWarning

yep started seeing that

File: pandas/core/generic.py [hunk 1]
```diff
@@ -11887,3 +11862,23 @@ def _doc_params(cls):
     The required number of valid values to perform the operation. If fewer than
     ``min_count`` non-NA values are present the result will be NA.
 """
+
+
+def _align_as_utc(
```
Comment (jreback) [inline pandas/core/generic.py:11867]: prob should move somewhere else, but can be in the future

**Pull request merged**